test/unit/kubernetes-deploy/kubernetes_resource/daemon_set_test.rb

```diff
@@ -38,22 +38,105 @@ def test_deploy_passes_when_updated_available_does_match
       "observedGeneration": 2,
     }
 
-    ds_template = build_ds_template(status: status)
-    ds = build_synced_ds(template: ds_template)
+    ds_template = build_ds_template(filename: 'daemon_set.yml', status: status)
+    ds = build_synced_ds(ds_template: ds_template)
     assert_predicate(ds, :deploy_succeeded?)
   end
 
+  def test_deploy_passes_when_ready_pods_for_one_node
+    status = {
+      "desiredNumberScheduled": 1,
+      "updatedNumberScheduled": 1,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', generation: 2, status: status)
+    pod_template = build_pod_template(filename: 'daemon_set_pod.yml')
+    node_template = build_node_template(filename: 'node.yml')
+    ds = build_synced_ds(ds_template: ds_template, pod_templates: pod_template, node_templates: node_template)
+    assert_predicate(ds, :deploy_succeeded?)
+  end
+
+  def test_deploy_passes_when_ready_pods_but_node_added
+    status = {
+      "desiredNumberScheduled": 1,
+      "updatedNumberScheduled": 1,
+      "numberReady": 0,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', generation: 2, status: status)
+    pod_template = build_pod_template(filename: 'daemon_set_pod_not_ready.yml')
+    node_template = build_node_template(filename: 'node.yml')
+    ds = build_synced_ds(ds_template: ds_template, pod_templates: pod_template, node_templates: node_template)
+    refute_predicate(ds, :deploy_succeeded?)
+
+    node_added_status = {
+      "desiredNumberScheduled": 3,
+      "updatedNumberScheduled": 3,
+      "numberReady": 1,
+    }
+    ds_template = build_ds_template(filename: 'daemon_set.yml', generation: 2, status: node_added_status)
+    pod_template = build_pod_template(filename: 'daemon_set_pod.yml')
+    node_template = build_node_template(filename: 'nodes.yml')
+    ds = build_synced_ds(ds_template: ds_template, pod_templates: pod_template, node_templates: node_template)
```

Comment:
> I think that test_deploy_passes_when_ready_pods_but_node_added should cover this, but i wasn't sure how I could use the same daemonset instance, though I'm not totally sure it matters.

> Can you think of a way to reuse the daemonset instance and update the status transparently? Ultimately to test logic around cache I think that's needed.

This is the line that needs to change--you need to manually do this method's job on your old instance:

```ruby
    stub_kind_get("DaemonSet", items: [ds_template]) # the new one from L75
    stub_kind_get("Pod", items: [ready_pod_template]) # the new one from L76
    stub_kind_get("Node", items: three_node_templates) # maybe you can cheat and return the same one three times instead of having a separate fixture for this one too?
    ds.sync(build_resource_cache) # this puts the new data into the existing instance
```

Also, this is your regression test, so when you're done with it, make sure it fails if you revert your business logic change.